src/eom-application.c

```diff
@@ -332,24 +332,11 @@ eom_application_get_file_window (EomApplication *application, GFile *file)
 static void
 eom_application_show_window (EomWindow *window, gpointer user_data)
 {
-	guint32 timestamp = GPOINTER_TO_UINT (user_data);
-
-	/* set the proper interaction time on the window.
-	 * Fall back to roundtripping to the X server when we
-	 * don't have the timestamp, e.g. when launched from
-	 * terminal. We also need to make sure that the window
-	 * has been realized otherwise it will not work. lame.
-	 */
 	if (!gtk_widget_get_realized (GTK_WIDGET (window)))
 		gtk_widget_realize (GTK_WIDGET (window));
 
-	if (timestamp <= 0)
-		timestamp = gdk_x11_get_server_time (gtk_widget_get_window (GTK_WIDGET (window)));
```

Comment:
Shouldn't this be simply guarded against running X11?  Removing it without a good rationale sounds dangerous, because I would imagine there was a reason this got introduced in the first place.

Reply:
I replaced the deleted code with the gtk_window_present_with_time function, is this incorrect?

Reply:
I would think that it's a good replacement for the `gdk_x11_window_set_user_time()` call, but not for handling of `timestamp == 0`.  The [docs for `gtk_window_present_with_time()`](https://developer.gnome.org/gtk3/stable/GtkWindow.html#gtk-window-present-with-time) do _not_ suggest they handle `timestamp == 0` in a specific manner themselves.

Reply:
Also, it'd be worth keeping at least part of the comment above the `gtk_widget_realize()` call

Reply:
Well, all this hackisch X-code was added with mate-1.14 to fix issue https://github.com/mate-desktop/eom/issues/119.
https://github.com/mate-desktop/eom/commit/1b65a062fc01e15c4a461463a055b90d8cb3235e
Before this comment we use 
```
gtk_window_present_with_time (GTK_WINDOW (window),
                              GPOINTER_TO_UINT (user_data));
```
And latest eog master use the same code that we used before https://github.com/mate-desktop/eom/commit/1b65a062fc01e15c4a461463a055b90d8cb3235e
see https://gitlab.gnome.org/GNOME/eog/blob/master/src/eog-application.c#L602
So, i am thinking https://github.com/mate-desktop/eom/issues/119 should be fixed in a proper way without adding X code. 

Btw. i will test if current status of PR causes the focus problem again.

Reply:
I can't reproduce https://github.com/mate-desktop/eom/issues/119 with this PR under X11 and using `gtk_window_present_with_time`.
Tested with marcos default settings and opening nearly 60 images from caja or from terminal when a eom window was already open.
Keep in mind that https://github.com/mate-desktop/eom/issues/119 did occours under Mate with gtk+-2 and before port to GtkApplication.
So, i am thinking using `gtk_window_present_with_time` is fine.
But you guys knows really more about coding than me ;)